html/janus.js

```diff
@@ -44,10 +44,109 @@ Janus.isExtensionEnabled = function() {
 	}
 };
 
+Janus.useDefaultDependencies = function (fetchFn, adapterObj, promiseCls) {
+	var f = fetchFn || fetch;
+	var p = promiseCls || Promise;
+	return {
+		isArray: function(arr) { return Array.isArray(arr); },
+		checkJanusExtendsion: function() { return document.querySelector('#janus-extension-installed') !== null; },
+		webRTCAdapter: adapterObj || adapter,
+		httpAPICall: function(url, options) {
+			var fetchOptions = {method: options.verb, cache: 'no-cache'};
+			if(options.withCredentials !== undefined) {
+				fetchOptions.credentials = options.withCredentials === true ? 'include' : (options.withCredentials ? options.withCredentials : 'omit');
+			}
+			if(options.body !== undefined) {
+				fetchOptions.body = JSON.stringify(options.body);
+			}
+
+			var fetching = f(url, fetchOptions).then(function(response) {
+				if(response.ok) {
+					// minor optimisation: no need to parse response body if there is no success handler to do anything with it.
+					if(typeof(options.success) === typeof(Janus.noop)) {
+						return response.json().then(function(result) {
+							// success!
+							options.success(result);
+						}).catch(function(error) {
+							// JSON parsing error
+							if(typeof(options.error) === typeof(Janus.noop)) {
+								options.error(response.statusText, error);
+							}
+						});
+					}
+				}
+				else {
+					// API error (HTTP status code says so)
+					if(typeof(options.error) === typeof(Janus.noop)) {
+						options.error(response.statusText, response.status);
+					}
+				}
+			}).catch(function(error) { // network error
+				if(typeof(options.error) === typeof(Janus.noop)) {
+					options.error('<< network error >>', error);
+				}
+			});
+
+			/*
+			 * fetch() does not natively support timeouts.
+			 * Work around this by starting a timeout manually, and racing it agains the fetch() to see which thing resolves first.
+			 */
+			if(options.timeout !==  undefined) {
+				var timeoutError = 'request timeout';
+				var timeout = new p(function(resolve, reject) {
+					setTimeout(reject, options.timeout, timeoutError);
+				}).catch(function(error) { // timeout error
+					if(typeof(options.error) === typeof(Janus.noop)) {
+						options.error('<< network error >>', error);
+					}
+				});
+
+				fetching = p.race([fetching, timeout]);
+			}
+			return fetching;
+		}
+	}
+};
+
+Janus.useOldDependencies = function (jQueryObj, adapterObj) {
+	var jq = jQueryObj || $;
```

Comment:
I think
```javascript
var jq = jQueryObj || jQuery;
```
would be less brittle here? In the off case that someone was using other libs that expose $.

Reply:
I think that should be safe, will do.